spark/v3.4/spark/src/main/java/org/apache/iceberg/spark/source/BaseBatchReader.java

```diff
@@ -61,9 +66,12 @@ protected CloseableIterable<ColumnarBatch> newBatchIterable(
       SparkDeleteFilter deleteFilter) {
     switch (format) {
       case PARQUET:
-        return newParquetIterable(inputFile, start, length, residual, idToConstant, deleteFilter);
+        return newParquetIterable(
+            inputFile, keyMetadata, start, length, residual, idToConstant, deleteFilter);
```

Comment:
Can this pass the `EncryptedInputFile` instead? That would avoid needing to pass both separately.